cloudinit/config/schema.py

```diff
@@ -1476,39 +1518,39 @@ def handle_schema_args(name, args):
     else:
         instance_data_path = paths.get_runpath("instance_data_sensitive")
     if args.config_file:
-        config_files = ((args.schema_type, args.config_file),)
+        config_files: Tuple[Tuple[str, str], ...] = (
+            (args.schema_type or "cloud-config", args.config_file),
+        )
     else:
         if os.getuid() != 0:
             error(
                 "Unable to read system userdata or vendordata as non-root"
                 " user. Try using sudo.",
                 sys_exit=True,
             )
-        userdata_file = paths.get_ipath("cloud_config")
+        userdata_file = get_processed_or_fallback_path(
+            paths, "cloud_config", "userdata_raw"
+        )
         if not userdata_file:
             error(
                 "Unable to obtain user data file. No instance data available",
                 sys_exit=True,
             )
-            return  # Helps typing
-
-        # Prefer raw user-data.txt when processed cloud-config is empty and
-        # raw user-data.txt is not because processed cloud-config.txt will
-        # not be written in cases where user-data header is not supported.
-        try:
-            if os.stat(userdata_file).st_size == 0:
-                raw_userdata_file = paths.get_ipath("userdata_raw")
-                if os.stat(raw_userdata_file).st_size:
-                    userdata_file = raw_userdata_file
-        except FileNotFoundError:
-            # Error handling on absent userdata_file below
-            pass
-
         config_files = (("user-data", userdata_file),)
-        supplemental_config_files = (
-            ("vendor-data", paths.get_ipath("vendor_cloud_config")),
-            ("vendor2-data", paths.get_ipath("vendor2_cloud_config")),
-            ("network-config", paths.get_ipath("network_config")),
+        supplemental_config_files: Tuple[Tuple[str, str], ...] = (
```

Comment:
Great suggestions here. While it's a bit of a refactor of the logic, now is as good a time as any because a followup PR will make these even harder when we are dealing with network schema version 2. I've added Enums and an InstanceDataType named tuple to better capture the magic strings and clarify specific SchemaTypes associated with user-data, vendor*data and network-config.